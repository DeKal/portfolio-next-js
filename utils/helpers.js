import { INIT_SELECTED_PAGE } from '~/consts/initState'
import $ from 'jquery'

export const initHashLocation = () => {
  if (location.hash) {
    const offset = $(location.hash).offset()
    window.scrollTo(offset.left, offset.top)
  }
}

export const isSafari = () => {
  var userAgent = navigator.userAgent.toLowerCase()
  if (userAgent.indexOf('safari') != -1) {
    // Safari agent will not have Chrome
    if (userAgent.indexOf('chrome') > -1) {
      return false
    } else {
      return true
    }
  }
}

export const setJqueryScrollEvent = () => {
  $('a').on('click', function(event) {
    // Make sure this.hash has a value before overriding default behavior
    if (this.hash) {
      // Prevent default anchor click behavior
      event.preventDefault()
      // Store hash
      var hash = this.hash

      $('html, body').animate(
        {
          scrollTop: $(hash).offset().top
        },
        400,
        function() {
          // Add hash (#) to URL when done scrolling (default click behavior)
          window.location.hash = hash
        }
      )
    } // End if
  })
}

export const cleanUpScrollEvent = () => {
  $('a').unbind()
}

export const getSelectedSection = () => {
  if (location.hash.length !== 0) {
    const selectedPage = location.hash.substr(1)
    return selectedPage
  }
  return INIT_SELECTED_PAGE
}

export const isChristmasDay = () => {
  const today = new Date()
  const month = today.getMonth() + 1
  const isDecember = month == 12
  const day = today.getDate()
  const isChristMasTime = 20 <= day <= 30

  return isDecember && isChristMasTime
}

export const isNewYearComing = () => {
  const today = new Date()
  const day = today.getDate()
  const isFirstTwoWeek = 1 <= day <= 16
  const month = today.getMonth() + 1
  const isJanuary = month == 1

  return isJanuary && isFirstTwoWeek
}
