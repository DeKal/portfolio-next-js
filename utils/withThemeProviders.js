import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import defaultTheme from '~/views/theme'

export const shallowWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

export const mountWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}
