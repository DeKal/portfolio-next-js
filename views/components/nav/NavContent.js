import React, { useState } from 'react'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import NavLogo from '~/views/components/nav/NavLogo'
import NavItem from '~/views/components/nav/NavItem'
import { PAGE_LIST } from '~/consts/pages'
import { INIT_SELECTED_PAGE } from '~/consts/initState'

const NavContent = ({ isShowNavContent }) => {
  const [selectedPage, setSelectedPage] = useState(INIT_SELECTED_PAGE)

  return (
    <Container isShowNavContent={isShowNavContent} className="second-nav">
      <div className="container">
        <NavLogo />
        <NavMenu className="menu">
          <ul>
            {PAGE_LIST.map(page => (
              <NavLi key={page}>
                <NavItem
                  page={page}
                  isSelected={selectedPage === page}
                  setSelectedPage={setSelectedPage}
                />
              </NavLi>
            ))}
          </ul>
        </NavMenu>
      </div>
    </Container>
  )
}

NavContent.propTypes = {
  isShowNavContent: PropTypes.bool
}

const Container = styled.div`
  position: relative;
  min-height: ${props => props.theme.navContentMinHeight};
  background-color: ${props => props.theme.colors.colorLight};
  margin-top: ${props =>
    props.isShowNavContent
      ? props.theme.navHeaderHeight
      : `-${props.theme.navHeaderHeight}`};
  box-shadow: ${props => props.theme.extra.boxShadow};
  transition: all 0.5s;

  @media screen and (max-width: 800px) {
    margin-top: -100%;
  }
`
const NavMenu = styled.nav`
  text-align: right;

  @media screen and (max-width: 800px) {
    text-align: center;
  }
`
const NavLi = styled.li`
  display: inline-block;
  margin-left: 18px;

  @media screen and (max-width: 800px) {
    margin: 0 9px;
    display: block;
    border-bottom: 1px dashed ${props => props.theme.colors.colorDarken};
  }
`
export default NavContent
