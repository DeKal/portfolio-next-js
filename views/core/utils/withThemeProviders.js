import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import homeTheme from '~/theme/theme'
import baseTheme from '~/theme/baseTheme'

/* istanbul ignore next */
export const shallowWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

/* istanbul ignore next */
export const shallowWithDivingTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children).dive()
}

/* istanbul ignore next */
export const mountWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}

/* istanbul ignore next */
export const shallowWithBaseTheme = (children, theme = baseTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}
