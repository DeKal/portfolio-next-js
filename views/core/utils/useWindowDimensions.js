import { useState, useEffect } from 'react'
import { getWindowDimensions } from '~/views/core/utils/helpers'

/* istanbul ignore next */
const useWindowDimensions = () => {
  const [windowDimensions, setWindowDimensions] = useState(
    getWindowDimensions()
  )

  useEffect(() => {
    /* istanbul ignore next */
    function handleResize() {
      setWindowDimensions(getWindowDimensions())
    }

    window.addEventListener('resize', handleResize)
    /* istanbul ignore next */
    return () => window.removeEventListener('resize', handleResize)
  }, [])

  return windowDimensions
}

export default useWindowDimensions
