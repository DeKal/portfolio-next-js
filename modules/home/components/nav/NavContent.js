import React, { useState, useEffect } from 'react'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import NavLogo from '~/modules/home/components/nav/NavLogo'
import NavItem from '~/modules/home/components/nav/NavItem'
import MobileContainer from '~/modules/home/components/common/container/MobileContainer'
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'
import { PAGE_LIST } from '~/modules/home/consts/pages'
import {
  getSelectedSection,
  setJqueryScrollEvent,
  cleanUpScrollEvent,
  isSafari,
  initHashLocation
} from '~/modules/core/utils/helpers'
import useWindowDimensions from '~/modules/core/utils/useWindowDimensions'

const NavContent = ({ isShowNavContent }) => {
  const [selectedPage, setSelectedPage] = useState(getSelectedSection())
  const { width } = useWindowDimensions()
  useEffect(() => {
    const handleHashChange = () => {
      setSelectedPage(getSelectedSection())
    }
    window.addEventListener('hashchange', handleHashChange)
    initHashLocation()
    /*
     * Currently safari doesnt support smooth scrolling with scroll-behavior
     * This is a workaround by jquery for the case in Safari
     */
    if (isSafari()) {
      setJqueryScrollEvent()
    }

    return function cleanup() {
      window.removeEventListener('hashchange', handleHashChange, false)
      if (isSafari()) {
        cleanUpScrollEvent()
      }
    }
  }, [])

  return (
    <NavContainer isShowNavContent={isShowNavContent}>
      <MobileContainer>
        <Row>
          <Col lg={2}>
            <NavLogo />
          </Col>
          <MobileCol fluid="true">
            <NavMenu>
              <NavUl>
                {PAGE_LIST.map(
                  page =>
                    (width > 768 || page.isShowOnMobile) && (
                      <NavLi key={page.name}>
                        <NavItem
                          page={page}
                          isMobile={width <= 768}
                          isSelected={selectedPage === page.cmpName}
                          setSelectedPage={setSelectedPage}
                        />
                      </NavLi>
                    )
                )}
              </NavUl>
            </NavMenu>
          </MobileCol>
        </Row>
      </MobileContainer>
    </NavContainer>
  )
}

NavContent.propTypes = {
  isShowNavContent: PropTypes.bool
}

const NavContainer = styled.div`
  position: relative;
  min-height: ${props => props.theme.navContentMinHeight};
  background-color: ${props => props.theme.colors.colorLight};
  margin-top: ${props =>
    props.isShowNavContent
      ? props.theme.navHeaderHeight
      : `-${props.theme.navHeaderHeight}`};
  box-shadow: ${props => props.theme.extra.boxShadow};
  transition: all 0.5s;

  @media screen and (max-width: 768px) {
    margin-top: ${props =>
      props.isShowNavContent
        ? props.theme.navHeaderHeight
        : props.theme.navPosWhenHiding};
    position: fixed;
    min-height: 0;
    width: 100%;
    bottom: 0;
    background: white;
    border-top: 1px solid ${props => props.theme.colors.colorAccent};
  }
`
const MobileCol = styled(Col)`
  @media screen and (max-width: 768px) {
    padding: 0 !important;
  }
`
const NavMenu = styled.nav`
  text-align: right;

  @media screen and (max-width: 768px) {
    text-align: center;
  }
`

const NavUl = styled.ul`
  @media screen and (max-width: 768px) {
    display: flex;
    justify-content: space-evenly;
  }
`
const NavLi = styled.li`
  display: inline-block;
  margin-left: 18px;

  @media screen and (max-width: 768px) {
    display: flex;
    margin: 8px;
    font-size: 13px;
  }
`
export default NavContent
