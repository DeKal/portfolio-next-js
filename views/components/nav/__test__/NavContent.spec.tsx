import * as React from 'react'
import { mountWithTheme, shallowWithTheme } from '~/utils/withThemeProviders'
import NavContent from '~/views/components/nav/NavContent'

describe('Nav Content', () => {
  it('should render without throwing an error', function() {
    const wrap = shallowWithTheme(<NavContent />)
    expect(wrap).toMatchSnapshot()
  })

  it('should render with isShowNavContent enable', function() {
    const wrap = mountWithTheme(<NavContent isShowNavContent />)
    expect(wrap).toMatchSnapshot()
  })

  it('should render with isShowNavContent disable', function() {
    const wrap = mountWithTheme(<NavContent isShowNavContent={false} />)
    expect(wrap).toMatchSnapshot()
  })

  it('test hashChange when clicking to nav content', function() {
    const wrap = mountWithTheme(<NavContent isShowNavContent={false} />)
    wrap.find('[data-test-id="nav-link"]').first().simulate('click')
  })
})
