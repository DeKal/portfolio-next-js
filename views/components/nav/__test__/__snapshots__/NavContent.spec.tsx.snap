// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Content should render with isShowNavContent disable 1`] = `
<NavContent
  isShowNavContent={false}
>
  <NavContent__NavContainer
    isShowNavContent={false}
  >
    <div
      className="NavContent__NavContainer-sc-1bgb07m-0 eHTNdM"
    >
      <MobileContainer>
        <Container
          className="MobileContainer-sc-14bpgy4-0 cmmyDP"
          fluid={false}
        >
          <div
            className="MobileContainer-sc-14bpgy4-0 cmmyDP container"
          >
            <ForwardRef
              noGutters={false}
            >
              <div
                className="row"
              >
                <Col
                  md={2}
                >
                  <div
                    className="col-md-2"
                  >
                    <NavLogo>
                      <NavLogo__Logo
                        href="https://github.com/DeKal/portfolio-next-js"
                      >
                        <a
                          className="NavLogo__Logo-sc-1822jsu-0 bmvzrf"
                          href="https://github.com/DeKal/portfolio-next-js"
                        >
                          <NavLogo__LogoText>
                            <span
                              className="NavLogo__LogoText-sc-1822jsu-1 dQmdA-D"
                            >
                              JD
                            </span>
                          </NavLogo__LogoText>
                        </a>
                      </NavLogo__Logo>
                    </NavLogo>
                  </div>
                </Col>
                <Col
                  fluid="true"
                >
                  <div
                    className="col"
                    fluid="true"
                  >
                    <NavContent__NavMenu>
                      <nav
                        className="NavContent__NavMenu-sc-1bgb07m-1 kgJvPe"
                      >
                        <ul>
                          <NavContent__NavLi
                            key="Home"
                          >
                            <li
                              className="NavContent__NavLi-sc-1bgb07m-2 gnuOKH"
                            >
                              <NavItem
                                isSelected={true}
                                page="Home"
                                setSelectedPage={[Function]}
                              >
                                <NavItem__SelectedContainer>
                                  <a
                                    className="NavItem__Container-a312fv-0 NavItem__SelectedContainer-a312fv-1 fhutBo"
                                  >
                                    <Translate
                                      id="Home"
                                    >
                                      <WrappedTranslate
                                        context={
                                          Object {
                                            "activeLanguage": undefined,
                                            "addTranslation": [Function],
                                            "addTranslationForLanguage": [Function],
                                            "defaultLanguage": undefined,
                                            "ignoreTranslateChildren": false,
                                            "initialize": [Function],
                                            "languages": Array [],
                                            "renderToStaticMarkup": false,
                                            "setActiveLanguage": [Function],
                                            "translate": [Function],
                                          }
                                        }
                                        id="Home"
                                      >
                                        Missing translationId: Home for language: \${ languageCode }
                                      </WrappedTranslate>
                                    </Translate>
                                  </a>
                                </NavItem__SelectedContainer>
                              </NavItem>
                            </li>
                          </NavContent__NavLi>
                          <NavContent__NavLi
                            key="About"
                          >
                            <li
                              className="NavContent__NavLi-sc-1bgb07m-2 gnuOKH"
                            >
                              <NavItem
                                isSelected={false}
                                page="About"
                                setSelectedPage={[Function]}
                              >
                                <NavItem__Container
                                  data-test-id="nav-link"
                                  href="#About"
                                  onClick={[Function]}
                                >
                                  <a
                                    className="NavItem__Container-a312fv-0 dSWZyj"
                                    data-test-id="nav-link"
                                    href="#About"
                                    onClick={[Function]}
                                  >
                                    <Translate
                                      id="About"
                                    >
                                      <WrappedTranslate
                                        context={
                                          Object {
                                            "activeLanguage": undefined,
                                            "addTranslation": [Function],
                                            "addTranslationForLanguage": [Function],
                                            "defaultLanguage": undefined,
                                            "ignoreTranslateChildren": false,
                                            "initialize": [Function],
                                            "languages": Array [],
                                            "renderToStaticMarkup": false,
                                            "setActiveLanguage": [Function],
                                            "translate": [Function],
                                          }
                                        }
                                        id="About"
                                      >
                                        Missing translationId: About for language: \${ languageCode }
                                      </WrappedTranslate>
                                    </Translate>
                                  </a>
                                </NavItem__Container>
                              </NavItem>
                            </li>
                          </NavContent__NavLi>
                          <NavContent__NavLi
                            key="Experiences"
                          >
                            <li
                              className="NavContent__NavLi-sc-1bgb07m-2 gnuOKH"
                            >
                              <NavItem
                                isSelected={false}
                                page="Experiences"
                                setSelectedPage={[Function]}
                              >
                                <NavItem__Container
                                  data-test-id="nav-link"
                                  href="#Experiences"
                                  onClick={[Function]}
                                >
                                  <a
                                    className="NavItem__Container-a312fv-0 dSWZyj"
                                    data-test-id="nav-link"
                                    href="#Experiences"
                                    onClick={[Function]}
                                  >
                                    <Translate
                                      id="Experiences"
                                    >
                                      <WrappedTranslate
                                        context={
                                          Object {
                                            "activeLanguage": undefined,
                                            "addTranslation": [Function],
                                            "addTranslationForLanguage": [Function],
                                            "defaultLanguage": undefined,
                                            "ignoreTranslateChildren": false,
                                            "initialize": [Function],
                                            "languages": Array [],
                                            "renderToStaticMarkup": false,
                                            "setActiveLanguage": [Function],
                                            "translate": [Function],
                                          }
                                        }
                                        id="Experiences"
                                      >
                                        Missing translationId: Experiences for language: \${ languageCode }
                                      </WrappedTranslate>
                                    </Translate>
                                  </a>
                                </NavItem__Container>
                              </NavItem>
                            </li>
                          </NavContent__NavLi>
                          <NavContent__NavLi
                            key="Projects"
                          >
                            <li
                              className="NavContent__NavLi-sc-1bgb07m-2 gnuOKH"
                            >
                              <NavItem
                                isSelected={false}
                                page="Projects"
                                setSelectedPage={[Function]}
                              >
                                <NavItem__Container
                                  data-test-id="nav-link"
                                  href="#Projects"
                                  onClick={[Function]}
                                >
                                  <a
                                    className="NavItem__Container-a312fv-0 dSWZyj"
                                    data-test-id="nav-link"
                                    href="#Projects"
                                    onClick={[Function]}
                                  >
                                    <Translate
                                      id="Projects"
                                    >
                                      <WrappedTranslate
                                        context={
                                          Object {
                                            "activeLanguage": undefined,
                                            "addTranslation": [Function],
                                            "addTranslationForLanguage": [Function],
                                            "defaultLanguage": undefined,
                                            "ignoreTranslateChildren": false,
                                            "initialize": [Function],
                                            "languages": Array [],
                                            "renderToStaticMarkup": false,
                                            "setActiveLanguage": [Function],
                                            "translate": [Function],
                                          }
                                        }
                                        id="Projects"
                                      >
                                        Missing translationId: Projects for language: \${ languageCode }
                                      </WrappedTranslate>
                                    </Translate>
                                  </a>
                                </NavItem__Container>
                              </NavItem>
                            </li>
                          </NavContent__NavLi>
                          <NavContent__NavLi
                            key="Certifications"
                          >
                            <li
                              className="NavContent__NavLi-sc-1bgb07m-2 gnuOKH"
                            >
                              <NavItem
                                isSelected={false}
                                page="Certifications"
                                setSelectedPage={[Function]}
                              >
                                <NavItem__Container
                                  data-test-id="nav-link"
                                  href="#Certifications"
                                  onClick={[Function]}
                                >
                                  <a
                                    className="NavItem__Container-a312fv-0 dSWZyj"
                                    data-test-id="nav-link"
                                    href="#Certifications"
                                    onClick={[Function]}
                                  >
                                    <Translate
                                      id="Certifications"
                                    >
                                      <WrappedTranslate
                                        context={
                                          Object {
                                            "activeLanguage": undefined,
                                            "addTranslation": [Function],
                                            "addTranslationForLanguage": [Function],
                                            "defaultLanguage": undefined,
                                            "ignoreTranslateChildren": false,
                                            "initialize": [Function],
                                            "languages": Array [],
                                            "renderToStaticMarkup": false,
                                            "setActiveLanguage": [Function],
                                            "translate": [Function],
                                          }
                                        }
                                        id="Certifications"
                                      >
                                        Missing translationId: Certifications for language: \${ languageCode }
                                      </WrappedTranslate>
                                    </Translate>
                                  </a>
                                </NavItem__Container>
                              </NavItem>
                            </li>
                          </NavContent__NavLi>
                          <NavContent__NavLi
                            key="Contacts"
                          >
                            <li
                              className="NavContent__NavLi-sc-1bgb07m-2 gnuOKH"
                            >
                              <NavItem
                                isSelected={false}
                                page="Contacts"
                                setSelectedPage={[Function]}
                              >
                                <NavItem__Container
                                  data-test-id="nav-link"
                                  href="#Contacts"
                                  onClick={[Function]}
                                >
                                  <a
                                    className="NavItem__Container-a312fv-0 dSWZyj"
                                    data-test-id="nav-link"
                                    href="#Contacts"
                                    onClick={[Function]}
                                  >
                                    <Translate
                                      id="Contacts"
                                    >
                                      <WrappedTranslate
                                        context={
                                          Object {
                                            "activeLanguage": undefined,
                                            "addTranslation": [Function],
                                            "addTranslationForLanguage": [Function],
                                            "defaultLanguage": undefined,
                                            "ignoreTranslateChildren": false,
                                            "initialize": [Function],
                                            "languages": Array [],
                                            "renderToStaticMarkup": false,
                                            "setActiveLanguage": [Function],
                                            "translate": [Function],
                                          }
                                        }
                                        id="Contacts"
                                      >
                                        Missing translationId: Contacts for language: \${ languageCode }
                                      </WrappedTranslate>
                                    </Translate>
                                  </a>
                                </NavItem__Container>
                              </NavItem>
                            </li>
                          </NavContent__NavLi>
                        </ul>
                      </nav>
                    </NavContent__NavMenu>
                  </div>
                </Col>
              </div>
            </ForwardRef>
          </div>
        </Container>
      </MobileContainer>
    </div>
  </NavContent__NavContainer>
</NavContent>
`;

exports[`Nav Content should render with isShowNavContent enable 1`] = `
<NavContent
  isShowNavContent={true}
>
  <NavContent__NavContainer
    isShowNavContent={true}
  >
    <div
      className="NavContent__NavContainer-sc-1bgb07m-0 bnWvqA"
    >
      <MobileContainer>
        <Container
          className="MobileContainer-sc-14bpgy4-0 cmmyDP"
          fluid={false}
        >
          <div
            className="MobileContainer-sc-14bpgy4-0 cmmyDP container"
          >
            <ForwardRef
              noGutters={false}
            >
              <div
                className="row"
              >
                <Col
                  md={2}
                >
                  <div
                    className="col-md-2"
                  >
                    <NavLogo>
                      <NavLogo__Logo
                        href="https://github.com/DeKal/portfolio-next-js"
                      >
                        <a
                          className="NavLogo__Logo-sc-1822jsu-0 bmvzrf"
                          href="https://github.com/DeKal/portfolio-next-js"
                        >
                          <NavLogo__LogoText>
                            <span
                              className="NavLogo__LogoText-sc-1822jsu-1 dQmdA-D"
                            >
                              JD
                            </span>
                          </NavLogo__LogoText>
                        </a>
                      </NavLogo__Logo>
                    </NavLogo>
                  </div>
                </Col>
                <Col
                  fluid="true"
                >
                  <div
                    className="col"
                    fluid="true"
                  >
                    <NavContent__NavMenu>
                      <nav
                        className="NavContent__NavMenu-sc-1bgb07m-1 kgJvPe"
                      >
                        <ul>
                          <NavContent__NavLi
                            key="Home"
                          >
                            <li
                              className="NavContent__NavLi-sc-1bgb07m-2 gnuOKH"
                            >
                              <NavItem
                                isSelected={true}
                                page="Home"
                                setSelectedPage={[Function]}
                              >
                                <NavItem__SelectedContainer>
                                  <a
                                    className="NavItem__Container-a312fv-0 NavItem__SelectedContainer-a312fv-1 fhutBo"
                                  >
                                    <Translate
                                      id="Home"
                                    >
                                      <WrappedTranslate
                                        context={
                                          Object {
                                            "activeLanguage": undefined,
                                            "addTranslation": [Function],
                                            "addTranslationForLanguage": [Function],
                                            "defaultLanguage": undefined,
                                            "ignoreTranslateChildren": false,
                                            "initialize": [Function],
                                            "languages": Array [],
                                            "renderToStaticMarkup": false,
                                            "setActiveLanguage": [Function],
                                            "translate": [Function],
                                          }
                                        }
                                        id="Home"
                                      >
                                        Missing translationId: Home for language: \${ languageCode }
                                      </WrappedTranslate>
                                    </Translate>
                                  </a>
                                </NavItem__SelectedContainer>
                              </NavItem>
                            </li>
                          </NavContent__NavLi>
                          <NavContent__NavLi
                            key="About"
                          >
                            <li
                              className="NavContent__NavLi-sc-1bgb07m-2 gnuOKH"
                            >
                              <NavItem
                                isSelected={false}
                                page="About"
                                setSelectedPage={[Function]}
                              >
                                <NavItem__Container
                                  data-test-id="nav-link"
                                  href="#About"
                                  onClick={[Function]}
                                >
                                  <a
                                    className="NavItem__Container-a312fv-0 dSWZyj"
                                    data-test-id="nav-link"
                                    href="#About"
                                    onClick={[Function]}
                                  >
                                    <Translate
                                      id="About"
                                    >
                                      <WrappedTranslate
                                        context={
                                          Object {
                                            "activeLanguage": undefined,
                                            "addTranslation": [Function],
                                            "addTranslationForLanguage": [Function],
                                            "defaultLanguage": undefined,
                                            "ignoreTranslateChildren": false,
                                            "initialize": [Function],
                                            "languages": Array [],
                                            "renderToStaticMarkup": false,
                                            "setActiveLanguage": [Function],
                                            "translate": [Function],
                                          }
                                        }
                                        id="About"
                                      >
                                        Missing translationId: About for language: \${ languageCode }
                                      </WrappedTranslate>
                                    </Translate>
                                  </a>
                                </NavItem__Container>
                              </NavItem>
                            </li>
                          </NavContent__NavLi>
                          <NavContent__NavLi
                            key="Experiences"
                          >
                            <li
                              className="NavContent__NavLi-sc-1bgb07m-2 gnuOKH"
                            >
                              <NavItem
                                isSelected={false}
                                page="Experiences"
                                setSelectedPage={[Function]}
                              >
                                <NavItem__Container
                                  data-test-id="nav-link"
                                  href="#Experiences"
                                  onClick={[Function]}
                                >
                                  <a
                                    className="NavItem__Container-a312fv-0 dSWZyj"
                                    data-test-id="nav-link"
                                    href="#Experiences"
                                    onClick={[Function]}
                                  >
                                    <Translate
                                      id="Experiences"
                                    >
                                      <WrappedTranslate
                                        context={
                                          Object {
                                            "activeLanguage": undefined,
                                            "addTranslation": [Function],
                                            "addTranslationForLanguage": [Function],
                                            "defaultLanguage": undefined,
                                            "ignoreTranslateChildren": false,
                                            "initialize": [Function],
                                            "languages": Array [],
                                            "renderToStaticMarkup": false,
                                            "setActiveLanguage": [Function],
                                            "translate": [Function],
                                          }
                                        }
                                        id="Experiences"
                                      >
                                        Missing translationId: Experiences for language: \${ languageCode }
                                      </WrappedTranslate>
                                    </Translate>
                                  </a>
                                </NavItem__Container>
                              </NavItem>
                            </li>
                          </NavContent__NavLi>
                          <NavContent__NavLi
                            key="Projects"
                          >
                            <li
                              className="NavContent__NavLi-sc-1bgb07m-2 gnuOKH"
                            >
                              <NavItem
                                isSelected={false}
                                page="Projects"
                                setSelectedPage={[Function]}
                              >
                                <NavItem__Container
                                  data-test-id="nav-link"
                                  href="#Projects"
                                  onClick={[Function]}
                                >
                                  <a
                                    className="NavItem__Container-a312fv-0 dSWZyj"
                                    data-test-id="nav-link"
                                    href="#Projects"
                                    onClick={[Function]}
                                  >
                                    <Translate
                                      id="Projects"
                                    >
                                      <WrappedTranslate
                                        context={
                                          Object {
                                            "activeLanguage": undefined,
                                            "addTranslation": [Function],
                                            "addTranslationForLanguage": [Function],
                                            "defaultLanguage": undefined,
                                            "ignoreTranslateChildren": false,
                                            "initialize": [Function],
                                            "languages": Array [],
                                            "renderToStaticMarkup": false,
                                            "setActiveLanguage": [Function],
                                            "translate": [Function],
                                          }
                                        }
                                        id="Projects"
                                      >
                                        Missing translationId: Projects for language: \${ languageCode }
                                      </WrappedTranslate>
                                    </Translate>
                                  </a>
                                </NavItem__Container>
                              </NavItem>
                            </li>
                          </NavContent__NavLi>
                          <NavContent__NavLi
                            key="Certifications"
                          >
                            <li
                              className="NavContent__NavLi-sc-1bgb07m-2 gnuOKH"
                            >
                              <NavItem
                                isSelected={false}
                                page="Certifications"
                                setSelectedPage={[Function]}
                              >
                                <NavItem__Container
                                  data-test-id="nav-link"
                                  href="#Certifications"
                                  onClick={[Function]}
                                >
                                  <a
                                    className="NavItem__Container-a312fv-0 dSWZyj"
                                    data-test-id="nav-link"
                                    href="#Certifications"
                                    onClick={[Function]}
                                  >
                                    <Translate
                                      id="Certifications"
                                    >
                                      <WrappedTranslate
                                        context={
                                          Object {
                                            "activeLanguage": undefined,
                                            "addTranslation": [Function],
                                            "addTranslationForLanguage": [Function],
                                            "defaultLanguage": undefined,
                                            "ignoreTranslateChildren": false,
                                            "initialize": [Function],
                                            "languages": Array [],
                                            "renderToStaticMarkup": false,
                                            "setActiveLanguage": [Function],
                                            "translate": [Function],
                                          }
                                        }
                                        id="Certifications"
                                      >
                                        Missing translationId: Certifications for language: \${ languageCode }
                                      </WrappedTranslate>
                                    </Translate>
                                  </a>
                                </NavItem__Container>
                              </NavItem>
                            </li>
                          </NavContent__NavLi>
                          <NavContent__NavLi
                            key="Contacts"
                          >
                            <li
                              className="NavContent__NavLi-sc-1bgb07m-2 gnuOKH"
                            >
                              <NavItem
                                isSelected={false}
                                page="Contacts"
                                setSelectedPage={[Function]}
                              >
                                <NavItem__Container
                                  data-test-id="nav-link"
                                  href="#Contacts"
                                  onClick={[Function]}
                                >
                                  <a
                                    className="NavItem__Container-a312fv-0 dSWZyj"
                                    data-test-id="nav-link"
                                    href="#Contacts"
                                    onClick={[Function]}
                                  >
                                    <Translate
                                      id="Contacts"
                                    >
                                      <WrappedTranslate
                                        context={
                                          Object {
                                            "activeLanguage": undefined,
                                            "addTranslation": [Function],
                                            "addTranslationForLanguage": [Function],
                                            "defaultLanguage": undefined,
                                            "ignoreTranslateChildren": false,
                                            "initialize": [Function],
                                            "languages": Array [],
                                            "renderToStaticMarkup": false,
                                            "setActiveLanguage": [Function],
                                            "translate": [Function],
                                          }
                                        }
                                        id="Contacts"
                                      >
                                        Missing translationId: Contacts for language: \${ languageCode }
                                      </WrappedTranslate>
                                    </Translate>
                                  </a>
                                </NavItem__Container>
                              </NavItem>
                            </li>
                          </NavContent__NavLi>
                        </ul>
                      </nav>
                    </NavContent__NavMenu>
                  </div>
                </Col>
              </div>
            </ForwardRef>
          </div>
        </Container>
      </MobileContainer>
    </div>
  </NavContent__NavContainer>
</NavContent>
`;

exports[`Nav Content should render without throwing an error 1`] = `
<NavContent__NavContainer>
  <MobileContainer>
    <ForwardRef
      noGutters={false}
    >
      <Col
        md={2}
      >
        <NavLogo />
      </Col>
      <Col
        fluid="true"
      >
        <NavContent__NavMenu>
          <ul>
            <NavContent__NavLi
              key="Home"
            >
              <NavItem
                isSelected={true}
                page="Home"
                setSelectedPage={[Function]}
              />
            </NavContent__NavLi>
            <NavContent__NavLi
              key="About"
            >
              <NavItem
                isSelected={false}
                page="About"
                setSelectedPage={[Function]}
              />
            </NavContent__NavLi>
            <NavContent__NavLi
              key="Experiences"
            >
              <NavItem
                isSelected={false}
                page="Experiences"
                setSelectedPage={[Function]}
              />
            </NavContent__NavLi>
            <NavContent__NavLi
              key="Projects"
            >
              <NavItem
                isSelected={false}
                page="Projects"
                setSelectedPage={[Function]}
              />
            </NavContent__NavLi>
            <NavContent__NavLi
              key="Certifications"
            >
              <NavItem
                isSelected={false}
                page="Certifications"
                setSelectedPage={[Function]}
              />
            </NavContent__NavLi>
            <NavContent__NavLi
              key="Contacts"
            >
              <NavItem
                isSelected={false}
                page="Contacts"
                setSelectedPage={[Function]}
              />
            </NavContent__NavLi>
          </ul>
        </NavContent__NavMenu>
      </Col>
    </ForwardRef>
  </MobileContainer>
</NavContent__NavContainer>
`;
